packages/rhf-mui/src/TextFieldElement.tsx

```diff
@@ -100,17 +100,16 @@ const TextFieldElement = forwardRef(function TextFieldElement<
         typeof transform?.input === 'function'
           ? transform.input
           : (value) => {
-              return value || ('' as TValue)
+              return value == null ? ('' as TValue) : value
             },
       output:
         typeof transform?.output === 'function'
           ? transform.output
           : (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
               const value = event.target.value
-              return (type === 'number' && value ? +value : value) as PathValue<
-                TFieldValues,
-                TName
-              >
+              return (
```

Comment:
I am afraid we have an edge case here:
if event.target.value is "" (empty string) then +"" == 0 thats why  I think we need to improve it in a different way